src/sage/schemes/elliptic_curves/kummer_line.py

```diff
@@ -0,0 +1,1765 @@
+            sage: K.j_invariant()
+            74
+        """
+        return self._curve.j_invariant()
+
+    @cached_method
+    def discriminant(self):
+        r"""
+        Compute the discriminant of the associated elliptic curve.
+
+        Cached method.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: K = KummerLine(E); K.discriminant()
+            44
+        """
+        return self._curve.discriminant()
+
+    def isogeny(self, xP):
+        r"""
+        Compute the isogeny with kernel ``xP``.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3]); K = KummerLine(E)
+            sage: P = E(76, 65); xP = K(P)
+            sage: Q = E(49, 61); xQ = K(Q)
+            sage: KummerLineIsogeny(xP)
+            Isogeny of degree 12 from Kummer Line of Elliptic Curve defined by y^2 = x^3 + 2*x + 3 over Finite Field of size 101 to Kummer Line of Elliptic Curve defined by y^2 = x^3 + 23*x + 73 over Finite Field of size 101
+        """
+
+        if xP._parent != self:
+            raise ValueError("point is not on the Kummer line")
+        return KummerLineIsogeny(xP)
+
+
+class KummerLine_weierstrass(KummerLine_generic):
+    r"""
+    Kummer line class for a short Weierstrass elliptic curve.
+
+    EXAMPLES::
+
+        sage: E = EllipticCurve(GF(101), [2, 3])
+        sage: K = KummerLine(E); K
+        Kummer Line of Elliptic Curve defined by y^2 = x^3 + 2*x + 3 over Finite Field of size 101
+    """
+
+    def __init__(self, curve):
+        r"""
+        Constructor for a Kummer line from a short Weierstrass elliptic curve.
+
+        INPUT: ``curve`` -- an elliptic curve in short Weierstrass form `y^2 = x^3 + a*x + b`
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: K = KummerLine(E); K
+            Kummer Line of Elliptic Curve defined by y^2 = x^3 + 2*x + 3 over Finite Field of size 101
+        """
+
+        super().__init__(curve)
+
+        ainvs = self._curve.a_invariants()
+        a, b = ainvs[3], ainvs[4]  # a = a4, b = a6
+
+        if ainvs != (0, 0, 0, a, b):
+            raise ValueError(
+                "input must be an elliptic curve in short Weierstrass form"
+            )
+
+        # Initialize variables
+        self._a = self._base_ring(a)
+        self._b = self._base_ring(b)
+
+        self._b2 = self._b + self._b
+        self._b4 = self._b2 + self._b2
+
+    def extract_weierstrass_constants(self):
+        r"""
+        Return the short Weierstrass coefficients `(a, b)` as a tuple.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: K = KummerLine(E); K.extract_weierstrass_constants()
+            (2, 3)
+        """
+        return self._a, self._b
+
+    def xDBL(self, XP, ZP):
+        r"""
+        Doubling formula for `x`-only coordinate on short Weierstrass curve.
+
+        Cost: ``2M + 5S + 3m0 + 8a``
```

Comment:
I know this is standard notation, but you might want to document the cost notation somewhere. Also, `xDBL`, `xADD`, `xDBLADD` aren't good method names, as they are not descriptive.

Also, why is `XP` and `ZP` provided separately? Shouldn't this be a point object or a tuple? Or am I confusing something?